diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5261f8ae1..03c69bad9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -135,7 +135,7 @@ find_package(ZLIB REQUIRED)

 # try to find an installed boost or install locally with conan
 set(boost_VERSION "1.71")
-find_package(Boost ${boost_VERSION} QUIET)
+# find_package(Boost ${boost_VERSION} QUIET)
 if (NOT Boost_FOUND)
   # bail if there's no conan installed
   message(STATUS "No compatible boost version detected, using conan...")
